From 7fc753f0cf7d2da025eda227fc2725c6523b3eee Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?R=C3=A9mi=20Bernon?= <rbernon@codeweavers.com>
Date: Fri, 12 Sep 2025 12:26:27 +0200
Subject: [PATCH] win32u: Remove some less useful flush_window_surfaces.

Now that wait_messages is doing it already, and as peek_message should
also call it occasionally.
---
 dlls/win32u/input.c   | 4 +---
 dlls/win32u/message.c | 2 --
 2 files changed, 1 insertion(+), 5 deletions(-)

diff --git a/dlls/win32u/input.c b/dlls/win32u/input.c
index c718bd1cbbc..b5ad3d1c027 100644
--- a/dlls/win32u/input.c
+++ b/dlls/win32u/input.c
@@ -796,9 +796,7 @@ static void check_for_events( UINT flags )
     };
     MSG msg;
 
-    if (!user_driver->pProcessEvents( flags ))
-        flush_window_surfaces( TRUE );
-
+    user_driver->pProcessEvents( flags );
     peek_message( &msg, &filter );
 }
 
diff --git a/dlls/win32u/message.c b/dlls/win32u/message.c
index f3e9f6858de..5fa5e10479d 100644
--- a/dlls/win32u/message.c
+++ b/dlls/win32u/message.c
@@ -3236,8 +3236,6 @@ static DWORD wait_objects( DWORD count, const HANDLE *handles, DWORD timeout,
 {
     assert( count );  /* we must have at least the server queue */
 
-    flush_window_surfaces( TRUE );
-
     if (!check_queue_masks( wake_mask, changed_mask ))
     {
         SERVER_START_REQ( set_queue_mask )
-- 
GitLab

